### ISSUE TITLE
Paginated response handling

### ISSUE THREAD

**Issue by jsamoocha** [hozn/stravalib#249]
There are many methods in `Client` (e.g., `get_activities()`) that return a `BatchedResultsIterator` instance. The Strava API parameter that manages how many results are returned per page is `per_page` (which, BTW, has a deprecation warning in the swagger.json but not in the published API docs).

However, none of the `stravalib` client methods support the `per_page` or `page` parameters that allow paginated results. To better match the Strava API and allow developers to customize paginated responses, wouldn't it be better if the client methods would have these `per_page` and `page` parameters as well?

In the short term, a quick-n-dirty workaround is to manually set these attributes on the `BatchedResultsIterator` instance returned by the client method.

**Comment by yihong0618** [comment]
I think we should consider that many users run `stravalib` in `CI`
and some of them did not lock the version in `requirements.txt`
If we have some break change we should be carefully.

**Comment by hozn** [comment]
Yeah, I think adding a new optional parameter would be fine (not breaking), but agree that we should be careful about not breaking API if the version is not changing major version numbers.

**Comment by lwasser** [comment]
hi @jsamoocha  - this is an older issue but still has some activity. Is this an ongoing issue or did you address this @jsamoocha with the big API restructure? Just lemme know as with the other issues . i can organize things we want to do vs closing older things that have been already addressed as it makes sense!

**Comment by jsamoocha** [comment]
This is not fixed yet.
